### ISSUE TITLE
Sever error (500) and "Programmer error"

### ISSUE THREAD

**Issue by charlesetc** [writeas/writefreely#221]
Hi! I got this log today after seeing a server error in my browser. I have just set up the blog and was trying to figure out what other people would see if I gave them the link so I opened it up in a new browser and got a server error. Let me know if I can help!
Best,
Charles

```
ERROR: 2019/12/07 04:29:22 collections.go:233: [PROGRAMMER ERROR] WARNING: Collection.hostName is empty! Federation and many other things will fail! If you're 
seeing this in the wild, please report this bug and let us know what you were doing just before this: https://github.com/writeas/writefreely/issues/new?templat
e=bug_report.md
ERROR: 2019/12/07 04:29:22 collections.go:233: [PROGRAMMER ERROR] WARNING: Collection.hostName is empty! Federation and many other things will fail! If you're 
seeing this in the wild, please report this bug and let us know what you were doing just before this: https://github.com/writeas/writefreely/issues/new?templat
e=bug_report.md
ERROR: 2019/12/07 04:29:22 collections.go:233: [PROGRAMMER ERROR] WARNING: Collection.hostName is empty! Federation and many other things will fail! If you're 
seeing this in the wild, please report this bug and let us know what you were doing just before this: https://github.com/writeas/writefreely/issues/new?templat
e=bug_report.md
2019/12/07 04:29:22 "GET /furniture-and-candles/edit" 200 3.224192ms "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:70.0) Gecko/20100101 Firefox/70.0"
2019/12/07 04:29:24 "POST /api/collections/charlespipin/posts/gamz0tg2ny" 200 2.917005ms "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:70.0) Gecko/20100101 Firef
ox/70.0"
2019/12/07 04:29:25 "GET /furniture-and-candles" 200 3.060187ms "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:70.0) Gecko/20100101 Firefox/70.0"
2019/12/07 04:29:26 "GET /" 200 8.126152ms "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:70.0) Gecko/20100101 Firefox/70.0"
2019/12/07 04:29:53 "GET /me/c/charlespipin" 200 1.510536ms "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:70.0) Gecko/20100101 Firefox/70.0"
2019/12/07 04:29:57 "POST /api/collections/charlespipin" 302 425.470656ms "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:70.0) Gecko/20100101 Firefox/70.0"
2019/12/07 04:29:57 "GET /me/c/charlespipin" 200 1.973216ms "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:70.0) Gecko/20100101 Firefox/70.0"
2019/12/07 04:30:01 "GET /me/c/charlespipin" 200 2.453504ms "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:70.0) Gecko/20100101 Firefox/70.0"
2019/12/07 04:30:03 "GET /me/posts/" 200 1.175054ms "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:70.0) Gecko/20100101 Firefox/70.0"
2019/12/07 04:30:04 "GET /me/new" 200 880.523µs "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:70.0) Gecko/20100101 Firefox/70.0"
ERROR: 2019/12/07 04:30:10 collections.go:681: Unable to render password-collection: html/template:password-collection.tmpl:29:14: no such template "user-supse
nded"
ERROR: 2019/12/07 04:30:10 handle.go:508: [Web handler] 500: html/template:password-collection.tmpl:29:14: no such template "user-supsended"
2019/12/07 04:30:10 Web internal error render
2019/12/07 04:30:10 "GET /" 500 4.361319ms "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:70.0) Gecko/20100101 Firefox/70.0"
2019/12/07 04:30:10 http: multiple response.WriteHeader calls
2019/12/07 04:30:10 "GET /favicon.ico" 200 312.806µs "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:70.0) Gecko/20100101 Firefox/70.0"
ERROR: 2019/12/07 04:30:12 collections.go:681: Unable to render password-collection: html/template:password-collection.tmpl:29:14: no such template "user-supse
nded"
ERROR: 2019/12/07 04:30:12 handle.go:508: [Web handler] 500: html/template:password-collection.tmpl:29:14: no such template "user-supsended"
2019/12/07 04:30:12 Web internal error render
2019/12/07 04:30:12 "GET /" 500 1.779703ms "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:70.0) Gecko/20100101 Firefox/70.0"
2019/12/07 04:30:12 http: multiple response.WriteHeader calls
2019/12/07 04:30:23 "GET /charlespiin" 302 475.625µs "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:70.0) Gecko/20100101 Firefox/70.0"
ERROR: 2019/12/07 04:30:23 collections.go:681: Unable to render password-collection: html/template:password-collection.tmpl:29:14: no such template "user-supse
nded"
ERROR: 2019/12/07 04:30:23 handle.go:508: [Web handler] 500: html/template:password-collection.tmpl:29:14: no such template "user-supsended"
2019/12/07 04:30:23 Web internal error render
2019/12/07 04:30:23 "GET /?g=charlespiin" 500 1.29674ms "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:70.0) Gecko/20100101 Firefox/70.0"
2019/12/07 04:30:23 http: multiple response.WriteHeader calls
```

**Comment by thebaer** [comment]
Hey, thanks for the report and including your logs! At first glance, this looks related to #216. 

Could you share a bit more about your app configuration? Is this a single-user instance (`single_user = true` in config.ini)?

**Comment by thebaer** [comment]
The `500` error is something new -- it should be fixed by #222.

**Comment by charlesetc** [comment]
Yep! `single_user = true`

Good to hear, thanks!
